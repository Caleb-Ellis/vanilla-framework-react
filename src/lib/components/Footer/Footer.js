import React from 'react';
import PropTypes from 'prop-types';
import './Footer.css';

const Footer = props => (
  <footer className="p-footer">
    { props.children }
  </footer>
);

Footer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Footer;
