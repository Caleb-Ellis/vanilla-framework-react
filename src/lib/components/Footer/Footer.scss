@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_footer';
@include vf-b-typography;
@include vf-b-links;
@include vf-p-footer;
