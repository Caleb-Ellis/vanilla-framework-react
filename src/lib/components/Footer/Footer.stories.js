import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Footer from './Footer';
import FooterNav from './FooterNav';
import FooterNavContainer from './FooterNavContainer';

storiesOf('Footer', module)
  .add('Default',
    withInfo('You can use this simple footer pattern to display simple HTML elements: lists, headings, columns, icons, buttons, etc. Centering is optional.')(() => (
      <Footer>
        <p>
          <strong>Footer</strong> by Me. The source code is licensed <a href="http://opensource.org/licenses/mit-license.php">MIT</a>.
          The website content is licensed <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY NC SA 4.0</a>.
        </p>
      </Footer>),
    ),
  )
  .add('With Links',
    withInfo('You can use this simple footer pattern to display copyright notices and a selection of links, using the FooterNavContainer and FooterNav sub-components.')(() => (
      <Footer>
        <p>&copy; 2017 Company name and logo are registered trademarks of Company Ltd.</p>
        <FooterNavContainer>
          <FooterNav title="Footer link 1" link="#" />
          <FooterNav title="Footer link 2" link="#" />
          <FooterNav title="Footer link 3" link="#" />
          <FooterNav title="Footer link 4" link="#" />
        </FooterNavContainer>
      </Footer>),
    ),
  );
