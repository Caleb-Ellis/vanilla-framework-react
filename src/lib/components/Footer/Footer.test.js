import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Footer from './Footer';
import FooterNavContainer from './FooterNavContainer';
import FooterNav from './FooterNav';

describe('Footer component', () => {
  it('should render basic HTML elements correctly', () => {
    const footer = ReactTestRenderer.create(
      <Footer>
        <h1>Heading 1</h1>
        <p>&copy; 2017 Company name and logo are registered trademarks of Company Ltd.</p>
        <ul>
          <li>List Item 1</li>
          <li>List Item 2</li>
        </ul>
      </Footer>,
    );
    const json = footer.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render FooterNavContainer and FooterNav components correctly', () => {
    const footer = ReactTestRenderer.create(
      <Footer>
        <p>&copy; 2017 Company name and logo are registered trademarks of Company Ltd.</p>
        <FooterNavContainer>
          <FooterNav title="Footer link 1" link="#" />
          <FooterNav title="Footer link 2" link="#" />
          <FooterNav title="Footer link 3" link="#" />
          <FooterNav title="Footer link 4" link="#" />
        </FooterNavContainer>
      </Footer>,
    );
    const json = footer.toJSON();
    expect(json).toMatchSnapshot();
  });
});
