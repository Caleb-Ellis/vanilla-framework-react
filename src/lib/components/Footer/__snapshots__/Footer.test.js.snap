// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render FooterNavContainer and FooterNav components correctly 1`] = `
<footer
  className="p-footer"
>
  <p>
    © 2017 Company name and logo are registered trademarks of Company Ltd.
  </p>
  <nav
    className="p-footer__nav"
  >
    <ul
      className="p-footer__links"
    >
      <li
        className="p-footer__item"
      >
        <a
          className="p-footer__link"
          href="#"
        >
          Footer link 1
        </a>
      </li>
      <li
        className="p-footer__item"
      >
        <a
          className="p-footer__link"
          href="#"
        >
          Footer link 2
        </a>
      </li>
      <li
        className="p-footer__item"
      >
        <a
          className="p-footer__link"
          href="#"
        >
          Footer link 3
        </a>
      </li>
      <li
        className="p-footer__item"
      >
        <a
          className="p-footer__link"
          href="#"
        >
          Footer link 4
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;

exports[`Footer component should render basic HTML elements correctly 1`] = `
<footer
  className="p-footer"
>
  <h1>
    Heading 1
  </h1>
  <p>
    © 2017 Company name and logo are registered trademarks of Company Ltd.
  </p>
  <ul>
    <li>
      List Item 1
    </li>
    <li>
      List Item 2
    </li>
  </ul>
</footer>
`;
